test/NuGet.Core.Tests/NuGet.Protocol.Tests/RawSearchResourceTests.cs

```diff
@@ -1,144 +0,0 @@
-// Copyright (c) .NET Foundation. All rights reserved.
```

Comment:
I wouldn't delete the whole class. 

There were some old tests unrelated to your work such as `RawSearchResource_SearchEncoding` for example (the 1st test).

Reply:
Exact same tests with PackageSearchResourceV3 implementation moved into PackageSearchResourceV3Tests.cs , so they're still covered.
I felt we're duplicate tests.

Reply:
They test different classes. Even if we mark `RawSearchResource` as obsolete, it's still part of our public API and should be supported as well as possible, including minimising the risk of regressions.

If we don't want to duplicate test code, we should have the test setup and test assertion code in one place, and execute it twice, once with `RawSearchResource` and once with `PackageSearchResourceV3`. In fact, doing it this way minimises risk of a future change changing behaviour in one of the two implementations.

Reply:
Yep, we should ensure our public APIs are tested even if they are unlikely to receive future updates.

Reply:
Ok. Restored it.